src/index.ts

```diff
@@ -14,11 +21,116 @@ export interface Env {
 	//
 	// Example binding to a Queue. Learn more at https://developers.cloudflare.com/queues/javascript-apis/
 	// MY_QUEUE: Queue;
+
+	DISCORD_TOKEN: string,
+	DISCORD_PUBLIC_KEY: string,
+	DISCORD_CLIENT_ID: string,
+	GITHUB_TOKEN?: string,
+	GUILD_ID?: string,
+	SUPPORT_CHANNEL?: string,
+	SUPPORT_AI_CHANNEL?: string,
+	SUPPORT_SQUAD_CHANNEL?: string,
+	STATS_SCHEDULE?: string,
+	SUPPORT_REDIRECT_SCHEDULE?: string,
+	ALGOLIA_INDEX?: string,
+	ALGOLIA_APP_ID?: string,
+	ALGOLIA_API_KEY?: string
+}
+
+export class JsonResponse extends Response {
+  constructor(body?: any, init?: ResponseInit) {
+    const jsonBody = JSON.stringify(body);
+    init = init || {
+      headers: {
+        'content-type': 'application/json;charset=UTF-8',
+      },
+    };
+    super(jsonBody, init);
+  }
 }
 
+const router = Router();
+
+router.get("/", async () => {
+	return new Response('Hello World!');
+})
+
+router.post("/", async (request, env: Env) => {
+
+	let interaction: APIBaseInteraction<InteractionType, any> | APIChatInputApplicationCommandInteraction | undefined;
+	const discordRequestData = await verifyDiscordRequest(
+		request,
+		env
+	);
+
+	interaction = discordRequestData.interaction;
+
+	if(!discordRequestData.isValid || !interaction)
+	{
+		return new Response('Bad request signature.', { status: 401 });
+	}
+
+	if(interaction.type == InteractionType.Ping)
+	{
+		return new JsonResponse({
+      type: InteractionResponseType.PONG,
+    });
+	}
+
+	if(interaction.type == InteractionType.ApplicationCommand)
+	{
+		interaction = interaction as APIChatInputApplicationCommandInteraction;
+		const interactionData: APIApplicationCommandInteractionData = interaction.data;
+
+		const command = commandList[interactionData.name];
+
+		if(command)
+		{
+			if(command.initialize)
+			{
+				if(!command.initialize(env))
+				{
+					return new Response("Internal error", {status: 500});
+				}
+			}
+
+			return await command.execute(interaction, env);
+		}
+
+		return new Response("Command not found", {status: 404});
+	}
+
+	if(interaction.type == InteractionType.ApplicationCommandAutocomplete)
+	{
+		interaction = interaction as APIApplicationCommandAutocompleteInteraction;
+		const interactionData: APIApplicationCommandInteractionData = interaction.data;
+
+		const command = commandList[interactionData.name];
+
+		if(command)
+		{
+			if(command.autocomplete)
+			{
+				if(command.initialize)
+				{
+					if(!command.initialize(env))
+					{
+						return new Response("Internal error", {status: 500});
+					}
+				}
+
+				return await command.autocomplete(interaction);
+			}
+		}
+		return new Response("Command not found", {status: 404});
+	}
+
+	return new Response("Yet to implement");
+})
```

Comment:
This is a bit to much nesting for my taste. Could we do this differently?

Reply:
We could, but for now I just want to get it all working